numba_dpex/experimental/launcher.py

```diff
@@ -160,8 +232,8 @@ def get_queue_ref_val(
         for arg_num, argty in enumerate(kernel_argtys):
             if isinstance(argty, DpnpNdArray):
                 llvm_val = kernel_args[arg_num]
-                datamodel = self._kernel_targetctx.data_model_manager.lookup(
-                    argty
+                datamodel = (
```

Comment:
Should it not work with either context? `DpctlSyclQueue` should be registered with both targets.

Reply:
It should, but we need to use cpu data type, since underlying datamodel is taken from boxing stage of `dpjit`, which is cpu side, not device. In other words if we would made changes to kernel datamodel, without changes to dpjit datamodel - it could result pointing to wrong memory address.

Reply:
Ok. makes sense